src/bindings/python/flux/uri/resolvers/lsf.py

```diff
@@ -0,0 +1,84 @@
+###############################################################
+# Copyright 2022 Lawrence Livermore National Security, LLC
+# (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+#
+# This file is part of the Flux resource manager framework.
+# For details, see https://github.com/flux-framework.
+#
+# SPDX-License-Identifier: LGPL-3.0
+###############################################################
+
+"""
+This module supports the Flux URI resolver for IBM Spectrum LSF.
+
+It can be used like:
+  `flux uri --remote lsf:<JOBID>`.
+
+"""
+
+import os
+import re
+import subprocess
+from pathlib import PurePath
+
+import yaml
+
+from flux.uri import URIResolverPlugin
+
+
+def lsf_find_compute_node(jobid):
+    """Figure out where the job is being run using JSON output from `bjobs -l`."""
+
+    sp = subprocess.run(
+        ["/opt/ibm/csm/bin/csm_allocation_query", "-j", str(jobid)], capture_output=True
+    )
+    try:
+        return str(yaml.safe_load(sp.stdout.decode("utf-8"))["compute_nodes"][0])
+    except:
+        raise ValueError("Unable to find a compute node attached to your job.")
+
+
+def lsf_job_pid(jobid, hostname):
+    """Resolve the pids of an LSF job using ssh to the compute node"""
+    sp = subprocess.run(
+        ["ssh", hostname, "ps", "-Ho", "pid,comm", "-u", os.getlogin()],
```

Comment:
This was my suggestion, and all my Python suggestions should be suspect :wink:. I just suggested the first thing I found online for the query "How to get the current username in Python".

Since the Python documentation says `getpass.getuser` should be prefered over `os.getlogin`, I would go with that. (With apologies to @wihobbs for the incorrect suggestion)